### PR TITLE
fix: make edit and delete comment button subtle

### LINKED ISSUE #3651: [feature request] Improve the display of the edit and delete buttons next to user's own comment
> This ticket is part of series of 7 tickets concluding a design task focusing on improving design and usability of comments feature. Here you can find the [Figma file](https://www.figma.com/board/GfQPMs7PyC7jbkymDTco2D/CP-%2F-Nested-Comments-Improvements?node-id=0-1&t=7sgMA4ZPYXAMb03A-1) for more context.

### **2/7 - Improve the display of the edit and delete buttons next to user's own comment**
---

**Is your feature request related to a problem? Please describe.**
The problem here is, that the height of the edit and delete buttons is increasing the spacing between the username and the text. This creates an inconsistency of spacing between user's own comments and other comments.

image

**Describe the solution you'd like**
A solution with consistent height of the wrapping div, that accounts for the size of the buttons regardless if they are there or not. The username should have `justify-content: flex-end;` 

image

Additionally, it would be good to have a less dominant version of the buttons as shown here. The issue of creating less dominant variant of the button component is prominent on several places throughout the platform and is discussed in this ticket #3655 . 

image

**Describe alternatives you've considered**
I first considered, making the buttons the size of the username component which is 25px, but that would be too small for mobile.

A menu with three dots is standard in other platforms, but I am not sure how big and complex of an implementation that would be, so I just propose this quick adjustment for now. However, the three dots menu was a consideration before and could be a good option, since the options will likely grow in the future (adding report function). The three dots would also perform better on mobile, not creating problems with stacking below.

**Additional context**
See the milestone

### PR DESCRIPTION
**Pull request by onim-at**
## PR Checklist

- [X] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)
- [ ] - Unit and e2e tests for the changes that have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix (fixes an issue)
- [ ] Feature (adds functionality)
- [X] Code style update
- [ ] Refactoring (no functional changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?
- Edit and Delete comment buttons use outlined variant
- Edit and Delete comment buttons increase the spacing between the username and the comment text

### Commit 1 [fix: make edit and delete comment button subtle](https://github.com/ONEARMY/community-platform/pull/3835/commits/48ac5c60f84ad41f4dbd764634c897ee3a6bf1f1)

Edit and Delete comment buttons use subtle variant:
image
on hover:
image

Updated  [button.ts](https://github.com/ONEARMY/community-platform/compare/master...onim-at:3651-comment-button?expand=1#diff-55ca3b7df8d945246670eada2c5f2cd5061d1ee57aafacd6c1374495054f53dd) subtle variant to be consistent with the changes done in #3675 

Updated [HideDiscussionContainer.tsx](https://github.com/ONEARMY/community-platform/compare/master...onim-at:3651-comment-button?expand=1#diff-6663f7705972fd1b641a8cd254882f3c09a31c1d3b37d167bd8a78472ea3ff34) to avoid regression due to the button component changes

### Commit 2 [fix: edit and delete comment position does not increase spacing](https://github.com/ONEARMY/community-platform/pull/3835/commits/037bffad53e4ffe07ed943acb8fce40486fe3f71)

Comment with edit and delete buttons height:
image

Comment without buttons height:
image

Bonus: the buttons have been moved to the right as per [figma design](https://www.figma.com/board/GfQPMs7PyC7jbkymDTco2D/CP-%2F-Nested-Comments-Improvements?node-id=0-1&node-type=CANVAS)

### Mobile view
<details>
<summary>Screenshot</summary>

image

</details>

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Git Issues

Closes #3651

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/common/HideDiscussionContainer.tsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const HideDiscussionContainer = ({
           textAlign: 'center',
           display: 'block',
           marginBottom: viewComments ? 2 : 0,
+          '&:hover': { bg: '#ececec' },
```
Comment (benfurber) [inline src/common/HideDiscussionContainer.tsx:51]: The theme colour 'offwhite' is set to #ececec, so best to use that please. i.e.:
```          '&:hover': { bg: 'offwhite' },```

Comment (benfurber) [reply src/common/HideDiscussionContainer.tsx:51]: Actually, as you've updated the styling for the `subtle` variant of buttons, how much of this custom styling is needed at all?

**Comment by benfurber** [comment]
Hey @onim-at, thanks for this! Looking good.

I've added a detailed code comment. Another thought, I'm not sure about the use of position on `packages/components/src/CommentItem/CommentItem.tsx`. I'd much rather the same effect was used with flex positioning as it'll allow easier changed in the future. I wonder if the `justifyContent` just needs to set to flex-end without the responsive array business?

**Comment by dalibormrska** [comment]
@onim-at Awesome work! I really like your thorough implementations, visually looks great, I'm excited to see it live!

File: src/common/HideDiscussionContainer.tsx (hunk 1 above)
Comment (onim-at) [reply src/common/HideDiscussionContainer.tsx:51]: Here a comparison of the HideDiscussionContainer button



<details> 
<summary>Current implementation with both hover and bg override. Same as what's in production today</summary>

https://github.com/user-attachments/assets/b8c98e77-cbcf-42cc-82b7-1069cb31607d
</details>

<details> 
<summary>Remove the hover override implemented in this PR</summary>

https://github.com/user-attachments/assets/27490f7b-6c79-4a68-9293-d68e33d8a08c
</details>

<details> 
<summary>Remove both hover and background override, rely only on the subtle button's colors</summary>

https://github.com/user-attachments/assets/6febc368-94df-49c4-8a9e-b1aea37c339b
</details>

The current implementation looks better to me, however I can see how implementing overrides may not be elegant
What do you think?

**Comment by onim-at** [comment]
@benfurber I'm not sure that would fix the height problem. I tried to play with the flex box but as the button size is bigger than the text on the left, it will always create more spacing. That's why the relative and absolute position, so that the button is independent of the flex and does not affect the total height
Is there something I'm missing about the flex-end?

File: src/common/HideDiscussionContainer.tsx (hunk 1 above)
Comment (benfurber) [reply src/common/HideDiscussionContainer.tsx:51]: Very clear, thanks. Yes I agree, your v1 is best.

**Comment by benfurber** [comment]
> @benfurber I'm not sure that would fix the height problem. I tried to play with the flex box but as the button size is bigger than the text on the left, it will always create more spacing. That's why the relative and absolute position, so that the button is independent of the flex and does not affect the total height Is there something I'm missing about the flex-end?

I realised it was going to take me three times as long to explain than to show. :) 
[This commit I added](https://github.com/ONEARMY/community-platform/pull/3835/commits/af7f8bb16301ae2dab48f344fc03ead23620d722) shows how to do it fully with flex.

**Pull request merged**